csireverseproxy/main_test.go

```diff
@@ -431,9 +573,95 @@ func TestServer_SAEventHandler(t *testing.T) {
 	} else {
 		fmt.Println("Secret Reverted Successfully")
 	}
+
+	log.Infof("End ServerSAEventHandlerTest")
 }
 
-func TestSAHTTPRequest(t *testing.T) {
+// Tests the secret handler, needs a running server
+func SecretHandlerTest(t *testing.T) {
+	log.Infof("Start SecretHandlerTest")
+	defer log.Infof("End SecretHandlerTest")
+
+	if os.Getenv(common.EnvReverseProxyUseSecret) == "false" {
+		log.Infof("SecretHandlerTest: Skipping test for config map case")
+		return
+	}
+	secret, err := readYAMLSecret(tmpSAConfigFile, common.TempConfigDir)
+	if err != nil {
+		t.Error("Failed to read config")
+		return
+	}
+
+	secret.ManagementServerConfig[0].Password = "test-password"
+	err = writeYAMLConfig(secret, tmpSAConfigFile, common.TempConfigDir)
+	// sleep for 10 seconds to allow the secret to be updated
+	time.Sleep(10 * time.Second)
+
+	secret.ManagementServerConfig[0].Password = "password"
+	err = writeYAMLConfig(secret, tmpSAConfigFile, common.TempConfigDir)
+	// sleep for 10 seconds to allow the secret to be updated
+	time.Sleep(10 * time.Second)
+}
+
+// Tests the params configmap handler of reverseproxy, needs a running server
+func CMParamsHandlerTest(t *testing.T) {
+	log.Infof("Start CMParamsHandlerTest")
+	defer log.Infof("End CMParamsHandlerTest")
+	if os.Getenv(common.EnvReverseProxyUseSecret) == "false" {
+		log.Infof("CMParamsHandlerTest: Skipping test for config map case")
+		return
+	}
+	cmp, err := readYAMLConfigParams(paramsConfigMapFile, common.TestConfigDir)
+	if err != nil {
+		t.Error("Failed to read config")
+		return
+	}
+
+	cmp.LogFormat = "json"
+	cmp.LogLevel = "debug"
+	err = writeYAMLConfig(cmp, paramsConfigMapFile, common.TestConfigDir)
+	// sleep for 10 seconds to allow the configmap to be updated
+	time.Sleep(10 * time.Second)
```

Comment:
Do we need to wait for 10 seconds or can we mock or reduce the time to wait? Adding multiple sleeps will increase time for executing our unit tests.

Reply:
10s was a bit of stretch. Sometimes I noticed it took a bit longer than a sec or two and I was probably being over-cautious. Changed it to 1s and ran the test several times. so far so good. I will keep an eye on it.